src/Blocks/BlockMushroom.h

```diff
@@ -32,14 +34,22 @@ class cBlockMushroomHandler:
 
 	virtual bool CanBeAt(cChunkInterface & a_ChunkInterface, const Vector3i a_RelPos, const cChunk & a_Chunk) override
 	{
-		if (a_RelPos.y <= 0)
+		if (!cChunkDef::IsValidHeight(a_RelPos.y - 1))
+		{
+			return false;
+		}
+
+		// Cannot be at too much light
+		NIBBLETYPE TotalLight = a_Chunk.GetSkyLightAltered(a_RelPos) + a_Chunk.GetBlockLight(a_RelPos);
+		if (TotalLight > MUSHROOM_MAX_LIGHT_THRESHOLD)
```

Comment:
Is this how mushrooms work? Usually `TotalLight` would be `std::max(SkyLightAltered, BlockLight)`.

Reply:
You're right, I used that at first, not sure why I changed it like that (light can be 15 at max anyways). note that denying the placement like that is still not like the vanilla behaviour. It drops when placed instead of just blocking the placement.